nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -193,7 +196,8 @@ def _process_example(self, example):
             total_ids += 1
         if self.add_sep:
             total_ids += 1
-        if self.add_eos:
+        # Only training need to consider eos token
+        if self.add_eos and self.tokens_to_generate == 0:
```

Comment:
are you using `self.tokens_to_generate == 0` to decide if we are in training phase or eval phase?

Reply:
Yes. Do you have a safer idea?